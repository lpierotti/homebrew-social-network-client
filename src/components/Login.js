import React from 'react'
import { connect } from 'react-redux'

export class Login extends React.Component {

	constructor() {
		super()
		this.state = {
			username: '',
			password: ''
		}
	}

	handleInputChange = (event) => {
		this.setState({
			[event.target.name]: event.target.value
		})
	}

	handleSubmit = (event) => {
		event.preventDefault()
		this.props.login(this.state)
	}

	render() {
		return (
			<form id='loginForm' className='login' onSubmit={this.handleSubmit}>
				<label>Username</label>
				<input id='email' onChange={this.handleInputChange} name='email' type='text' />
				<label>Password</label>
				<input id='password' onChange={this.handleInputChange} name='password' type='password' />
				<button id='submit'>Submit</button>
			</form>
		)
	}
}

function mapDispatchToProps(dispatch) {
	return {
		login: (userparams, history) => {
			dispatch(loginUser(userparams, history))
		}
	}
}

export default connect(mapDispatchToProps)(Login)