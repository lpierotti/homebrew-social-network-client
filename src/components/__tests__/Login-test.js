import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Login } from '../Login'
import renderer from 'react-test-renderer';


describe('Login Component', () => {
	let wrapper;
	const mockLoginfn = jest.fn();
	
	beforeEach(() => {
		wrapper = shallow(<Login login={mockLoginfn}/>)
	})

	it('renders correctly', () => {
		const tree = renderer.create(
			<Login />
		).toJSON();
		expect(tree).toMatchSnapshot();
	})
	
	it('should render without throwing an error', () => {
		expect(wrapper.exists(<form className='login'></form>)).toBe(true)
	})

	it('renders a email input', () => {
		expect(wrapper.find('#email').length).toEqual(1)
	})

	it('renders a password input', () => {
		expect(wrapper.find('#password').length).toEqual(1)
	})

	
	
	describe('Email input', () => {
		
		it('should respond to change event and change the state of the Login Component', () => {
			wrapper.find('#email').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}})
			expect(wrapper.state('email')).toEqual('blah@gmail.com')
		})
	})
	
	describe('Password input', () => {
		
		it('should respond to change event and change the state of the Login Component', () => {
			wrapper.find('#password').simulate('change', {target: {name: 'password', value: 'cats'}})
			expect(wrapper.state('password')).toEqual('cats')
		})
	})

	describe('When the form is submitted', () => {
		it('should call the mock login function', () => {
			wrapper.find('#loginForm').simulate('submit', {preventDefault() {}})
			expect(mockLoginfn.mock.calls.length).toBe(1)
		})
	})
	
})