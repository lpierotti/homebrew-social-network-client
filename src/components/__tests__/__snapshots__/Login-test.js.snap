// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<form
  className="login"
  id="loginForm"
  onSubmit={[Function]}
>
  <label>
    Username
  </label>
  <input
    id="email"
    name="email"
    onChange={[Function]}
    type="text"
  />
  <label>
    Password
  </label>
  <input
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <button
    id="submit"
  >
    Submit
  </button>
</form>
`;
